src/utils/viewer/viewer.cpp

```diff
@@ -201,17 +207,26 @@ Viewer::Viewer(const Arguments& arguments)
       Magnum::SceneGraph::AspectRatioPolicy::Extend);
 
   // Load navmesh if available
-  const std::string navmeshFilename = io::changeExtension(file, ".navmesh");
-  if (io::exists(navmeshFilename)) {
-    LOG(INFO) << "Loading navmesh from " << navmeshFilename;
-    pathfinder_->loadNavMesh(navmeshFilename);
-
-    const vec3f position = pathfinder_->getRandomNavigablePoint();
-    agentBodyNode_->setTranslation(Vector3(position));
+  if (file.compare(esp::assets::EMPTY_SCENE) != 0) {
+    const std::string navmeshFilename = io::changeExtension(file, ".navmesh");
+    if (io::exists(navmeshFilename) && !args.isSet("recompute-navmesh")) {
+      LOG(INFO) << "Loading navmesh from " << navmeshFilename;
+      pathfinder_->loadNavMesh(navmeshFilename);
+    } else {
+      // TODO: agent radius is not configured here
+      esp::nav::NavMeshSettings navMeshSettings;
+      navMeshSettings.setDefaults();
+      recomputeNavMesh(file, navMeshSettings);
+    }
   }
 
   // connect controls to navmesh if loaded
   if (pathfinder_->isLoaded()) {
+    // TODO: some scenes could have pathable roof polygons. We are not filtering
+    // those starting points here.
+    vec3f position = pathfinder_->getRandomNavigablePoint();
+    agentBodyNode_->setTranslation(Vector3(position));
```

Comment:
How are lines 227 and 228 related to this change?

Reply:
Without these, the agent's position is hardcoded in line 199 resulting in cases such as: 
![image](https://user-images.githubusercontent.com/1445143/71133541-b2bdd480-21af-11ea-8130-9a01c4b3f74f.png)

I guess this could be its own PR 😆

Reply:
You could make it a separate PR (generally preferred) or it should be mentioned explicitly in the commit message. Reviewing the history of the code (to locate a regression, etc) becomes difficult when the commit message and code do not agree.